etc/spack/defaults/darwin/modules.yaml

```diff
@@ -13,9 +13,4 @@
 # Per-user settings (overrides default and site settings):
 #   ~/.spack/modules.yaml
 # -------------------------------------------------------------------------
-modules:
-  prefix_inspections:
-    lib:
-      - DYLD_FALLBACK_LIBRARY_PATH
-    lib64:
-      - DYLD_FALLBACK_LIBRARY_PATH
```

Comment:
I understand your concern with `LD_LIBRARY_PATH`, but this issue does not arise with `DYLD_FALLBACK_LIBRARY_PATH`. On macOS, there are two env vars that control library selection at load-time: `DYLD_LIBRARY_PATH` and `DYLD_FALLBACK_LIBRARY_PATH`. The former acts just like `LD_LIBRARY_PATH` and can break system exes that aren't RPATHed. The latter acts more like RUNPATH and provides libraries to use only when the libraries aren't found in normal search locations. So I think it's safe to keep these here.

Reply:
Yeah, reverted that, it's just about linux now.